torchinductor/compile_fx.py

```diff
@@ -106,46 +106,29 @@ def compile_fx_inner(
     if cudagraphs is None:
         cudagraphs = config.triton.cudagraphs
 
-    if config.repro_level == 3:
-        dump_to_minify(gm, example_inputs)
-    try:
-        graph = GraphLowering(gm, num_dynamic_inputs=len(example_inputs))
-        with V.set_graph_handler(graph):
-            wrap(graph.run)(*example_inputs)
-            compiled_fn = wrap(graph.compile_to_fn())
-
-        # make sure it works, causes issues for mutation
-        # compiled_fn(*example_inputs)
-
-        if (
-            cudagraphs
-            and set(graph.device_types) == {"cuda"}
-            and not graph.mutated_inputs
-        ):
-            compiled_fn = cudagraphify(
-                compiled_fn, example_inputs, static_input_idxs=range(num_fixed)
-            )
-        elif cudagraphs:
-            if isinstance(cudagraphs, BoxedBool):
-                # Disable cudagraphs in the backwards pass too:
-                cudagraphs.value = False
-
-            if len(set(graph.device_types)) > 1:
-                log.warning("skipping cudagraphs due to multiple devices")
-            elif graph.mutated_inputs and set(graph.device_types) == {"cuda"}:
-                log.warning("skipping cudagraphs due to input mutation")
-
-        if config.repro_level > 0:
-            compiled_fn(*example_inputs)
-
-        return compiled_fn
-    except Exception:
-        if config.repro_level == 1:
-            dump_to_repro(gm, example_inputs)
-        elif config.repro_level == 2:
-            dump_to_minify(gm, example_inputs)
-
-        raise
+    graph = GraphLowering(gm, num_dynamic_inputs=len(example_inputs))
+    with V.set_graph_handler(graph):
+        wrap(graph.run)(*example_inputs)
+        compiled_fn = wrap(graph.compile_to_fn())
+
+    # make sure it works, causes issues for mutation
+    # compiled_fn(*example_inputs)
+
+    if cudagraphs and set(graph.device_types) == {"cuda"} and not graph.mutated_inputs:
```

Comment:
And cuda is available? Or checked upstream/downstream?

Reply:
If the graph contains tensors with "cuda types", then cuda is probably available :P